### PR TITLE
Feature/add cronjob and code cleanup

### PR DESCRIPTION
**Pull request by singularo**
* Lots of documentation updates where there were gaps/faults
* Readme updates
* Fix test so that it works again
* Add cron to api and tests
* Refactor volume handling in Client for more DRY
* Refactor get/delete functions for more DRY

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -49,143 +49,163 @@ class Client implements OpenShiftClientInterface {
    * @var array
    */
   protected $resourceMap = [
-    'secret' => [
+    'secret'                => [
       'create' => [
         'action' => 'POST',
-        'uri' => '/api/v1/namespaces/{namespace}/secrets',
+        'uri'    => '/api/v1/namespaces/{namespace}/secrets',
```
Comment (bart88) [inline src/Client.php]: Why 2 spaces ?

File: src/Client.php [hunk 2]
```diff
@@ -245,61 +270,71 @@ public function getApiVersion() {
...
       ];
     }
 
     try {
       $response = $this->guzzleClient->request($method, $uri, $requestOptions);
-    } catch (RequestException $exception) {
-      // @todo Handle errors.
+    }
+    catch (RequestException $exception) {
       return FALSE;
```
Comment (bart88) [inline src/Client.php]: Yay do nothing exception handling 🎉

File: src/Client.php (hunk 1 above)
Comment (singularo) [inline src/Client.php]: Lines up nicely in the editor. Meh. I'd prefer to leave it in as it looks a little nicer.

File: src/Client.php (hunk 2 above)
Comment (singularo) [inline src/Client.php]: Next MR ;-)

File: src/Client.php [hunk 3]
```diff
@@ -674,460 +687,541 @@ public function updateBuildConfig($name, $secret, $imagestream, $data) {
...
+      }
+    }
+    return [
+      'mounts' => $volume_mounts,
+      'config' => $volumes_config,
+    ];
+  }
+
+  /**
+   * Common function for very simple requeste to the api.
```
Comment (bart88) [inline src/Client.php]: Requesty requeste

File: README.md [hunk 4]
```diff
@@ -69,7 +69,7 @@ Manually create a mysql container in openshift, make note of the db name, userna
 Remove all objects created during tests : 
 ```bash
 # Assuming all the names for items created contain 'pied'
```
Comment (bart88) [inline README.md]: Changed pied to nothotdog 🌭  🚫

Comment (singularo) [inline README.md]: No. Nohotdog.

pied is the first part of the names in the example json, so matches them all.

**Comment by caseyfw** [comment]
I'm satisfied. Merge away.

**Pull request merged**